src/views/SsoSettings.vue

```diff
@@ -0,0 +1,483 @@
+
+export default defineComponent({
+  name: 'SsoSettings',
+  data() {
+    const provider = ['azure', 'okta']
+    const protocol = ['oauth2', 'saml2']
+    const groupNames = ['reblaze-events-viewer', 'reblaze-config-viewer', 'reblaze-config-editor', 'reblaze-org-admin']
+    return {
+      titles: DatasetsUtils.titles,
+      loadingCounter: 0,
+      showOAuth2: true,
+      showSaml2: false,
+      provider: provider,
+      protocol: protocol,
+      groupNames: groupNames,
+      removable: false,
+      ssoTitle: '',
+      jwtTokenName: '',
+      IdpGroup: '',
+      oauth2UserId: '',
+      oauth2UserSecret: '',
+      oauth2Issuer: '',
+      tokenExpiration: 0,
+      isSaveLoading: false,
+      ssoData: null as Sso,
+      threshold: 0,
+    }
+  },
+
+  watch: {
+    selectedBranch: {
+      handler: async function(val, oldVal) {
+        if ((this.$route.name as string).includes('Sso') && val && val !== oldVal) {
+          await this.loadSsoConfigs()
+        }
+      },
+      immediate: true,
+    },
+  },
+
+  computed: {
+    selectedBranch(): string {
+      return this.branchesStore.selectedBranchId
+    },
+
+    ...mapStores(useBranchesStore),
+  },
+  methods: {
+    setLoadingDocStatus(isLoading: boolean) {
+      if (isLoading) {
+        this.loadingCounter++
+      } else {
+        this.loadingCounter--
+      }
+    },
+
+    protocolSelector(event:any) {
+      if (event.target.value === 'oauth2') {
+        this.showOAuth2 = true
+        this.showSaml2 = false
+      } else {
+        this.showOAuth2 = false
+        this.showSaml2 = true
+      }
+    },
+
+    addRole() {
+      this.ssoData.roles.push({idp_group: '', reblaze_groups: []})
+    },
+
+    removeRole(index: number) {
+      this.ssoData.roles.splice(index, 1)
+    },
+
+    changeShown() {
+      this.ssoData.enabled = !this.ssoData.enabled
+    },
+
+    async loadSsoConfigs() {
+      this.setLoadingDocStatus(true)
+      this.isSaveLoading = true
+      const url = 'tools/sso/'
+      const response = await RequestsUtils.sendReblazeRequest({
+        methodName: 'GET',
+        url,
+        onFail: (error: any) => {
+          console.log('Error while attempting to load SSO', error)
+          this.ssoData = null
+          this.isSaveLoading = false
+          this.setLoadingDocStatus(true)
+        },
+      })
+      this.ssoData = response?.data || {}
+      this.normalizeSso()
+      this.setLoadingDocStatus(false)
+      this.isSaveLoading = false
+    },
+
+    normalizeSso() {
+      const basicSso: Sso = {
+        enabled: false,
+        title: '',
+        provider: 'okta',
+        protocol: 'oauth2',
+        roles_claim_key: '',
+        roles: [],
+        params: {
+          token_expiration_threshold: 600,
+          client_id: '',
+          client_secret: '',
+          issuer: '',
+          login_url: '',
+          tenantId: '',
+          entityId: '',
+          x509cert: '',
+        },
+      }
```

Comment:
We don't need to add every attribute to the object when normalizing it, simply adding objects and mandatory attributes would suffice

```suggestion
      const basicSso: Sso = {
        enabled: false,
        title: '',
        provider: 'okta',
        protocol: 'oauth2',
        roles: [],
        params: {},
      }
```

Reply:
Done